examples/examples/ws_notify_on_method_answered.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any
+// person obtaining a copy of this software and associated
+// documentation files (the "Software"), to deal in the
+// Software without restriction, including without
+// limitation the rights to use, copy, modify, merge,
+// publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software
+// is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice
+// shall be included in all copies or substantial portions
+// of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF
+// ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED
+// TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT
+// SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+// CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR
+// IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use std::net::SocketAddr;
+
+use jsonrpsee::core::client::ClientT;
+use jsonrpsee::proc_macros::rpc;
+use jsonrpsee::server::Server;
+use jsonrpsee::ws_client::WsClientBuilder;
+use jsonrpsee::{response_channel, rpc_params, ResponsePayloadV2};
+
+#[rpc(server, namespace = "state")]
+pub trait Rpc {
+	/// Async method call example.
+	#[method(name = "getKeys")]
+	fn storage_keys(&self) -> ResponsePayloadV2<'static, String>;
+}
+
+pub struct RpcServerImpl;
+
+impl RpcServer for RpcServerImpl {
+	fn storage_keys(&self) -> ResponsePayloadV2<'static, String> {
+		let (tx, rx) = response_channel();
+
+		tokio::spawn(async move {
+			rx.is_sent().await.unwrap();
+			println!("Method response to `state_getKeys` finished");
+		});
+
+		ResponsePayloadV2::result("ehheeheh".to_string()).notify_on_success(tx)
```

Comment:
Basically this is the new API which enforces that `rx` doesn't return until the message has been sent to internal websocket buffer